src/device_handle.rs

```diff
@@ -540,16 +540,16 @@ impl<T: UsbContext> DeviceHandle<T> {
     pub fn read_string_descriptor_ascii(&self, index: u8) -> crate::Result<String> {
         let mut buf = Vec::<u8>::with_capacity(255);
 
-        let buf_slice = unsafe { slice::from_raw_parts_mut(buf.as_mut_ptr(), buf.capacity()) };
-
-        let ptr = buf_slice.as_mut_ptr() as *mut c_uchar;
-        let len = buf_slice.len() as i32;
+        let ptr = buf.as_mut_ptr() as *mut c_uchar;
+        let len = buf.len() as i32;
 
         let res =
             unsafe { libusb_get_string_descriptor_ascii(self.handle.as_ptr(), index, ptr, len) };
 
         if res < 0 {
             return Err(error::from_libusb(res));
+        } else if res > len {
```

Comment:
This check is pointless. `libusb_get_string_descriptor_ascii` can't return result greater then len arg

Reply:
I though so. But I tend to be paranoid about things coming back from outside the app, particularly from C/FFI interfaces. And, since the next line is an unsafe call to set the buffer size to that value, it seems worthwhile to have the extra check.

Reply:
No. You must read documentation. In documentation say explicit 
"length - size of data buffer" No need check that.
Please remove that check